hydromt_wflow/wflow.py

```diff
@@ -1606,11 +1631,175 @@ def write_states(self):
         # raise NotImplementedError()
 
     def read_results(self):
-        """Read results at <root/?/> and parse to dict of xr.DataArray"""
+        """Read results at <root/?/> and parse to dict of xr.DataArray/xr.Dataset"""
         if not self._write:
             # start fresh in read-only mode
             self._results = dict()
-        # raise NotImplementedError()
+
+        # Read gridded netcdf (output section)
+        nc_fn = self.get_config("output.path", abs_path=True)
+        if nc_fn is not None and isfile(nc_fn):
+            self.logger.info(f"Read results from {nc_fn}")
+            ds = xr.open_dataset(nc_fn, chunks={"time": 30})
+            # TODO ? align coords names and values of results nc with staticmaps
+            self.set_results(ds, name="output")
+
+        # Read scalar netcdf (netcdf section)
+        ncs_fn = self.get_config("netcdf.path", abs_path=True)
+        if ncs_fn is not None and isfile(ncs_fn):
+            self.logger.info(f"Read results from {ncs_fn}")
+            ds = xr.open_dataset(ncs_fn, chunks={"time": 30})
+            self.set_results(ds, name="netcdf")
+
+        # Read csv timeseries (csv section)
+        csv_fn = self.get_config("csv.path", abs_path=True)
+        if csv_fn is not None and isfile(csv_fn):
```

Comment:
Would it be possible to make the csv result parsing a separate method outside the model class in e.g.: wflow_utils.py script (I did something similar for SFINCS)? That makes unit testing easier and also allows the user to use this method outside the model class.

Reply:
Good idea to add a wflow_utils.py script for this. I also started to do something similar for RIBASIM and delwaq :)